File: src/main/java/toast/client/modules/misc/CustomChat.java
package toast.client.modules.misc;

import net.minecraft.network.packet.c2s.play.ChatMessageC2SPacket;
import toast.client.ToastClient;
import toast.client.event.EventImpl;
import toast.client.event.events.network.EventPacketSent;
import toast.client.modules.Module;
import toast.client.utils.FancyChatUtil;

public class CustomChat extends Module {
    public static String suffix;
    public Boolean isMadeByCustomChat = false;

    public CustomChat() {
        super("CustomChat", "Custom chat messages", Category.MISC, -1);
        this.settings.addBoolean("Fancy Chat", false);
        this.settings.addBoolean("Commands", false);
        this.settings.addBoolean("Custom Suffix", false);
        this.settings.addMode("Separator", "None", "None", "Default", "Brackets");
        this.settings.addMode("Fancy chat type", "Retard", "Classic", "Retard", "Grammar", "Spaced");
    }

    @Override
    public void onEnable() {
        isMadeByCustomChat = true;
    }

    @EventImpl
    public void onEvent(EventPacketSent e) {
        if (mc.player == null) return;
        if (e.getPacket() instanceof ChatMessageC2SPacket) {
            if (!getBool("Custom Suffix")) {
                suffix = "ᴛᴏᴀѕᴛᴄʟɪᴇɴᴛ";
            }
            String packetMessage = ((ChatMessageC2SPacket) e.getPacket()).getChatMessage();
            if (packetMessage.startsWith(ToastClient.cmdPrefix) && !this.getBool("Commands")) return;
            if (packetMessage.startsWith("/") && !this.getBool("Commands")) return;
            if (!getBool("Fancy Chat")) {
                if (this.settings.getMode("Separator").equals("None")) {
                    String msg = FancyChatUtil.customSuffix(packetMessage, " ", suffix, "", false);
                    isMadeByCustomChat = !isMadeByCustomChat;
                    if (isMadeByCustomChat) return;
                    e.setCancelled(true);
                    mc.player.sendChatMessage(msg);
                }
                if (this.settings.getMode("Separator").equals("Default")) {
                    String msg = FancyChatUtil.customSuffix(packetMessage, " | ", suffix, "", false);
                    isMadeByCustomChat = !isMadeByCustomChat;
                    if (isMadeByCustomChat) return;
                    e.setCancelled(true);
                    mc.player.sendChatMessage(msg);
                }
                if (this.settings.getMode("Separator").equals("Brackets")) {
                    String msg = FancyChatUtil.customSuffix(packetMessage, " < ", suffix, " > ", true);
                    isMadeByCustomChat = !isMadeByCustomChat;
                    if (isMadeByCustomChat) return;
                    e.setCancelled(true);
                    mc.player.sendChatMessage(msg);
                }
            }
            if (getBool("Fancy Chat")) {
                if (this.settings.getMode("Fancy chat type").equals("Classic")) {
                    String msg = FancyChatUtil.classicFancy(packetMessage);
                    isMadeByCustomChat = !isMadeByCustomChat;
                    if (isMadeByCustomChat) return;
                    e.setCancelled(true);
//                    mc.player.sendChatMessage("msg1" +msg);
                    if (this.settings.getMode("Separator").equals("None")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " ", suffix, "", false);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                    if (this.settings.getMode("Separator").equals("Default")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " | ", suffix, "", false);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                    if (this.settings.getMode("Separator").equals("Brackets")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " < ", suffix, " > ", true);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                }
                if (this.settings.getMode("Fancy chat type").equals("Retard")) {
                    String msg = FancyChatUtil.retardChat(packetMessage);
                    isMadeByCustomChat = !isMadeByCustomChat;
                    if (isMadeByCustomChat) return;
                    e.setCancelled(true);
//                    mc.player.sendChatMessage("msg1" +msg);
                    if (this.settings.getMode("Separator").equals("None")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " ", suffix, "", false);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                    if (this.settings.getMode("Separator").equals("Default")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " | ", suffix, "", false);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                    if (this.settings.getMode("Separator").equals("Brackets")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " < ", suffix, " > ", true);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                }
                if (this.settings.getMode("Fancy chat type").equals("Spaced")) {
                    String msg = FancyChatUtil.spaces(packetMessage);
                    isMadeByCustomChat = !isMadeByCustomChat;
                    if (isMadeByCustomChat) return;
                    e.setCancelled(true);
//                    mc.player.sendChatMessage("msg1" +msg);
                    if (this.settings.getMode("Separator").equals("None")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " ", suffix, "", false);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                    if (this.settings.getMode("Separator").equals("Default")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " | ", suffix, "", false);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                    if (this.settings.getMode("Separator").equals("Brackets")) {
                        String msg2 = FancyChatUtil.customSuffix(msg, " < ", suffix, " > ", true);
                        isMadeByCustomChat = !isMadeByCustomChat;
                        if (isMadeByCustomChat) return;
                        e.setCancelled(true);
                        mc.player.sendChatMessage(msg2);
                    }
                }
                //Suffix Stuff

            }
        }
    }
}

File: src/main/kotlin/dev/toastmc/toastclient/impl/module/misc/CustomChat.kt
package dev.toastmc.toastclient.impl.module.misc

import dev.toastmc.toastclient.api.events.PacketEvent
import dev.toastmc.toastclient.api.managers.command.CommandManager
import dev.toastmc.toastclient.api.managers.module.Module
import dev.toastmc.toastclient.api.util.FancyChatUtil
import net.minecraft.network.packet.c2s.play.ChatMessageC2SPacket
import org.quantumclient.energy.Subscribe

object CustomChat : Module("CustomChat", Category.MISC) {

    var isMadeByCustomChat = false

    var suffix: String = "ᴛᴏᴀѕᴛᴄʟɪᴇɴᴛ"

    var suffixBool = bool("Suffix", true)
    var fancyChat = bool("FancyChat", false)
    var commands = bool("Commands", false)

    var fancyChatType = mode(
        "Fancy Chat Type",
        "Classic",
        "Retard", "Classic", "Spaced"
    )

    var separator = mode(
        "Separator",
        "None",
        "None", "Default", "Brackets"
    )

    override fun onEnable() {
        isMadeByCustomChat = true
    }

    @Subscribe
    fun on(event: PacketEvent.Receive) {
        if (mc.player == null) return
        if (event.packet is ChatMessageC2SPacket){
            val chatMessage = event.packet.chatMessage
            if ((chatMessage.startsWith(CommandManager.prefix) || chatMessage.startsWith("/")) && !commands.value) return
            suffix = when (separator.value){
                "None" -> " ᴛᴏᴀѕᴛᴄʟɪᴇɴᴛ"
                "Default" -> " | ᴛᴏᴀѕᴛᴄʟɪᴇɴᴛ"
                "Brackets" -> " < ᴛᴏᴀѕᴛᴄʟɪᴇɴᴛ >"
                else -> " ᴛᴏᴀѕᴛᴄʟɪᴇɴᴛ"
            }
            if(fancyChat.value){
                val fancyMsg = when (fancyChatType.value){
                    "Retard" -> FancyChatUtil.retardChat(chatMessage)
                    "Classic" -> FancyChatUtil.classicFancy(chatMessage)
                    "Fancy" -> FancyChatUtil.spaces(chatMessage)
                    else -> chatMessage
                }
                isMadeByCustomChat = !isMadeByCustomChat
                if(isMadeByCustomChat) return
                event.cancel()
                mc.player!!.sendChatMessage(fancyMsg + suffix, null)
            } else {
                isMadeByCustomChat = !isMadeByCustomChat
                if(isMadeByCustomChat) return
                event.cancel()
                mc.player!!.sendChatMessage(chatMessage + suffix, null)
            }
        }
    }
}